src/Middleware/StaticFiles/src/Helpers.cs

```diff
@@ -29,7 +29,7 @@ internal static bool IsGetOrHeadMethod(string method)
 
         internal static bool PathEndsInSlash(PathString path)
         {
-            return path.Value.EndsWith("/", StringComparison.Ordinal);
+            return path.HasValue && path.Value!.EndsWith("/", StringComparison.Ordinal);
```

Comment:
This seems like a bug. `PathSting.HasValue` is annotated with `MemberNotNullWhen` but it doesn't work here. @sharwell any idea why that might?

Reply:
![Null](https://user-images.githubusercontent.com/174281/96324054-8859c500-0fd4-11eb-8246-62eb152e66b8.gif)

Now in gif!